### ISSUE TITLE
Model not updating in controller 

### ISSUE THREAD

**Issue by dracostheblack** [adonespitogo/angular-base64-upload#51]
I have an issue where in my view i have 

    <input type="file" ng-model="attachment" class="ion-paperclip text-right float-right attachment" base-sixty-four-input onerror="errorHandler" onchange="fileChanged"/>

For one I have fileChanged defined in the controller and everytime I add a file that function comes up as undefined. 

For two I have a div with {{attachment}}  this is showing up correctly for when I add the file, but when I go to use that file in another function $scope.attachment, it's coming up as undefined. 

Any help would be appreciated.

Sean

**Comment by adonespitogo** [comment]
It's `on-change="fileChanged"`

**Comment by dracostheblack** [comment]
I fixed that, now i have         

fixed that but $scope.attachment is still undefined after I do a file upload

**Comment by adonespitogo** [comment]
Can you post your code and angular version?

**Comment by dracostheblack** [comment]
Angular is 1.3.6

In my controller all I have is. 

    $scope.attachment={};
    
    $scope.sendMessage = function () {
        console.log($scope.attachment.filename);
    };

    <div class="col col-10 text-right">
        <input type="file" ng-model="attachment" class="ion-paperclip text-right float-right attachment" base-sixty-four-input onerror="errorHandler" on-change="fileChanged"/>
    </div>

**Comment by adonespitogo** [comment]
Take a look at [this plunkr code](http://plnkr.co/edit/Abw4Yd1XRj4v68ZodOAT?p=preview) and see what's missing in your code.

**Comment by dracostheblack** [comment]
The only thing different I see is that i'm loading my controller from my module with stateprovider. 


        $stateProvider
            .state('inbox.landing', {
                url: '/landing',
                views: {
                    'bodyContent': {
                        templateUrl: 'js/modules/inbox/inbox.html',
                        controller: 'InboxCtrl'
                    }
                }

**Comment by adonespitogo** [comment]
Are you triggering `sendMessage()` within `fileChanged()`?

**Comment by dracostheblack** [comment]
No, i was just seeing what the value of the attachment, I don't need to use fileChanged if it will update the scope variable.

**Comment by dracostheblack** [comment]
Also validation is not firing for me either.

**Comment by lemetice** [comment]
Hi @dracostheblack , @adonespitogo 
Please how did you solde this issue of model not updating in controller?
 I am using Angular 1.5.7
I get **undefined** on debugging.

**Comment by sulmanweb** [comment]
What is the solution?
